src/thor/astar_bss.cc

```diff
@@ -42,8 +44,13 @@ AStarBSSAlgorithm::~AStarBSSAlgorithm() {
 // Clear the temporary information generated during path construction.
 void AStarBSSAlgorithm::Clear() {
   // Reduce edge labels capacity if it's more than limit
-  if (edgelabels_.size() > max_reserved_labels_count_) {
-    edgelabels_.resize(max_reserved_labels_count_);
+  if (!clear_reserved_memory_) {
+    if (edgelabels_.size() > max_reserved_labels_count_) {
+      edgelabels_.resize(max_reserved_labels_count_);
+      edgelabels_.shrink_to_fit();
+    }
+  } else {
+    edgelabels_.clear();
     edgelabels_.shrink_to_fit();
   }
```

Comment:
I think all of these can be simplified to this:

```suggestion
  edgelabels_.resize(clear_reserved_memory ? 0 : max_reserved_labels_count_);
  edgelabels_.shrink_to_fit();
```

see that below we are already clearing the labels no matter what. so the question is only really how much reservation do we want to have in the end? either we want 0 reservation when that boolean is set OR we want the max_reserved.